### DISCUSSION TITLE
Initialise a new bridge from a module

### DISCUSSION THREAD

**Discussion by jacobp100** [reactwg/react-native-new-architecture#25] [Q&A] [answered]
I'm trying to get a WebWorker package to work under the new arch. It used to work by initialising a new bridge, then transferring messages between.

Assuming this is still an okay approach, how do I get the module registry working?

https://github.com/jacobp100/react-native-webworker/blob/master/ios/RCTWebWorkerModule.m#L69-L72

I'm assuming I need the `#if RCT_NEW_ARCH_ENABLED` stuff from `AppDelegate.m`. Without it, I get a lot of errors about modules not being found.

However, when I do add it, I get the message `'React/RCTCxxBridgeDelegate.h' file not found`

Any pointers?

**Comment by cipolleschi** [comment]
Hi @jacobp100!
We recently shared [this repo](https://github.com/react-native-community/RNNewArchitectureLibraries) with the community.

Mir contains two branches, one that could guide you through the creation of a backward compatible TurboModule and one for a Fabric Component. In your case, the TurboModule will be more appropriate.

You should also have a look at the [The New Architecture](https://reactnative.dev/docs/new-architecture-intro) guide in the website. It contains a bunch of steps that are mandatory to prepare both your library and the receiving app. To support in that, there is this other [repo](https://github.com/cortinico/RNNewArchitectureApp) that contains sample migrations from the old architecture to the new one.

I hope these resources can help you!

**Reply by jacobp100** [reply]
Thanks for the links! I think my issue is quite specific - so it wasn't covered anywhere in the links.

**Reply by cipolleschi** [reply]
> However, when I do add it, I get the message 'React/RCTCxxBridgeDelegate.h' file not found

Do you get this message in the App that tries to use your module or in the Module itself?

**Reply by jacobp100** [reply]
I've got a setup here (iOS only so far)

https://github.com/jacobp100/react-native-webworker

My workflow is using the fabric example and editing the package contents under the Pods directory.

I'm suspecting I'm missing a dependency in my podspec, but cannot for the life of me figure out what dependency I need

**Comment by jacobp100** [comment, marked as answer]
This is resolved now - I had to add some private headers (https://github.com/jacobp100/react-native-webworker/blob/master/react-native-webworker.podspec#L28)
